### PR TITLE
feat: split JavaScript hash and encryption rules

### LINKED ISSUE #79: Javascript detect use of md5 when required
Expand md5 detection rules to find this case: https://github.com/appsecco/dvna/blob/master/core/authHandler.js#L3

### PR DESCRIPTION
**Pull request by elsapet**
## Description

* Split JavaScript weak encryption and weak password encryption rules into several new rules, broken down per library
* Also update description and recommendation copy for weak password encryption rules, since encryption is not recommended for passwords
* Extend MD5 hash patterns to include md5 library cases

Relates to #81 
Closes #79

## Checklist

- [x] I've added a snapshot that shows my rule works as expected.
- [x] My rule has adequate metadata to explain its use.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: javascript/shared/third_parties/crypto_init.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - javascript
+patterns:
+  - import $<!>$<_>$<...> from "crypto"
+  - pattern: |
+      $<METHOD>($<NAME>)
+    filters:
+      - variable: METHOD
+        values:
+          - require
+          - import
```
Comment (elsapet) [inline javascript/shared/third_parties/crypto_init.yml:12]: This isn't picking up `require` statements and I'm not sure why 🤔 e.g. 

```
const myCrypto = require("crypto")
myCrypto.createHmac(...)
```

File: javascript/shared/third_parties/crypto_init.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+type: shared
```
Comment (elsapet) [inline javascript/shared/third_parties/crypto_init.yml:1]: I'm not sure if these are strictly "third-party" - perhaps in a "lang" folder is better?

File: javascript/lang/weak_encryption_des.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+imports:
+  - javascript_shared_third_parties_cryptojs_init
+patterns:
+  - pattern: |
+      $<CRYPTOJS_INIT>.DES.encrypt($<OPTIONAL_DATA_TYPE>, $<_>)
+    filters:
+      - variable: CRYPTOJS_INIT
+        detection: crypto_js_init
```
Comment (didroe) [inline javascript/lang/weak_encryption_des.yml:8]:
```suggestion
        detection: javascript_shared_third_parties_cryptojs_init
```

File: javascript/shared/third_parties/crypto_init.yml (hunk 2 above)
Comment (didroe) [reply javascript/shared/third_parties/crypto_init.yml:1]: Yeah I think `lang` is best for NodeJS stdlib things.

File: javascript/shared/third_parties/crypto_init.yml (hunk 1 above)
Comment (elsapet) [reply javascript/shared/third_parties/crypto_init.yml]: ^ This was due to duplicate variable names in the shared rules (something we want to handle in the future). For now, we've added prefixes like `<RULE_NAME>_MY_VAR` to all variable names in the shared rules. 
Thanks to @didroe for the debugging on this 🙏

**Review by didroe — APPROVED** [review]

**Pull request merged**